tests/compilers/test_ast.py

```diff
@@ -404,6 +404,12 @@ def test_ast_tuple():
     assert type(code) == ast.Tuple
 
 
+def test_argument_destructuring():
+    """ Ensure argument destructuring works. """
+    can_compile("(fn [[a b]] (print a b))")
```

Comment:
Testing that something compiles isn't the same as testing that it behaves as intended. (For instance, the `can_compile("(fn [[a b]] (print a b))")` assertion passes on master.) Perhaps it may be worth adding an assertion that the AST object returned by `can_compile` is what we expect, or maybe adding a native test along the lines of&mdash;

```
(defn test-argument-destructuring []
  (defn function-with-destructured-args [a [b c] [d e f]]
    (, a b c d e f))
  (assert (= (, 1 2 3 4 5 6)
             (function-with-destructured-args 1 [2 3] [4 5 6]))))
```

(which errors against master with `NameError: name 'b' is not defined`, but passes against the `destruct-args` branch)?